server/src/main/java/org/candlepin/model/ConsumerTypeCurator.java

```diff
@@ -28,6 +28,33 @@ public ConsumerTypeCurator() {
         super(ConsumerType.class);
     }
 
+    /**
+     * Fetches the ConsumerType for the specified consumer. If the consumer does not have a defined
+     * type ID, or the type ID is invalid, this method throws an exception.
+     *
+     * @param consumer
+     *  The consumer for which to fetch the ConsumerType object
+     *
+     * @throws IllegalStateException
+     *  if the consumer has no defined type ID or the type ID is invalid
+     *
+     * @return
+     *  A ConsumerType instance for the specified consumer
+     */
+    public ConsumerType getConsumerType(Consumer consumer) {
+        ConsumerType type = null;
+
+        if (consumer != null && consumer.getTypeId() != null) {
+            type = this.find(consumer.getTypeId());
+
+            if (type == null) {
+                throw new IllegalStateException("consumer is not associated with a valid type: " + consumer);
+            }
+        }
+
+        return type;
```

Comment:
Debatable. My intent was to allow new, unmanaged consumers to be passed through without breaking things; but I can see an argument that getConsumerType should *always* return a valid type or explode. I tried to split the difference and make both approaches happy, but I'm not married to any one approach.

Reply:
The biggest reason that I brought this up was that there are places where we look up the type for the consumer and then check that type against type constants (type.equals(SOME_TYPE_CONSTANT) -- potential for an NPE -- though unlikely given the methods we are doing this in.

I'd like to see at least an exception thrown if the Consumer is null. To me, passing a null consumer to get a null type doesn't make sense, and would likely be an error. Also, we should update the javadoc on the method to describe the cases that null will be returned.

Reply:
Done. I've made getConsumerType far more restrictive. It will now throw an IllegalArgumentException if consumer is null or doesn't have any type at all, or an IllegalStateException if the type is invalid. The javadoc has been updated accordingly.